cmd/subnetcmd/deploy.go

```diff
@@ -104,26 +110,161 @@ func deploySubnet(cmd *cobra.Command, args []string) error {
+		if err := deployer.DeployToLocalNetwork(chain, chainGenesis); err != nil {
 			if deployer.BackendStartedHere() {
 				if innerErr := binutils.KillgRPCServerProcess(app); innerErr != nil {
 					app.Log.Warn("tried to kill the gRPC server process but it failed: %w", innerErr)
 				}
 			}
 		}
 		return err
+	case models.Fuji: // just make the switch pass
+		if keyName == "" {
+			return errors.New("this command requires the name of the private key")
+		}
+
+	case models.Mainnet: // just make the switch pass, fuij/main implementation is the same (for now)
 	default:
 		return errors.New("not implemented")
 	}
+
+	// prompt for control keys
+	controlKeys, cancelled, err := getControlKeys()
+	if err != nil {
+		return err
+	}
+	if cancelled {
+		ux.Logger.PrintToUser("User cancelled. No subnet deployed")
+		return nil
+	}
+
+	// prompt for threshold
+	var threshold uint32
+
+	if len(controlKeys) > 0 {
+		threshold, err = getThreshold(uint64(len(controlKeys)))
+		if err != nil {
+			return err
+		}
+	}
+
+	// deploy to public network
+	deployer := subnet.NewPublicSubnetDeployer(app, app.GetKeyPath(keyName), network)
+	subnetID, blockchainID, err := deployer.Deploy(controlKeys, threshold, chain, chainGenesis)
+	if err != nil {
+		return err
+	}
+
+	// update sidecar
+	// TODO: need to do something for backwards compatibility?
+	sidecar, err := app.LoadSidecar(chain)
+	if err != nil {
+		return err
+	}
+	sidecar.SubnetID = subnetID
+	sidecar.BlockchainID = blockchainID
+	return app.UpdateSidecar(&sidecar)
+}
+
+func getControlKeys() ([]string, bool, error) {
+	controlKeysPrompt := "Configure which addresses allow to add new validators"
+	// TODO: is this text ok?
+	noControlKeysPrompt := "You did not add any control key. This means anyone can add validators to your subnet. " +
+		"This is considered unsafe. Do you really want to continue?"
+
+	for {
+		// ask in a loop so that if some condition is not met we can keep asking
+		controlKeys, cancelled, err := controlKeysLoop(controlKeysPrompt)
+		if err != nil {
+			return nil, false, err
+		}
+		if cancelled {
+			return nil, cancelled, nil
+		}
+		if len(controlKeys) == 0 {
+			// we want to prompt if the user indeed wants to proceed without any control keys
+			yes, err := prompts.CaptureNoYes(noControlKeysPrompt)
+			if err != nil {
+				return nil, false, err
+			}
+			// the user confirms no control keys
+			if yes {
+				return controlKeys, false, nil
+			}
+			// otherwise we go back to the loop for asking
+		} else {
+			return controlKeys, false, nil
+		}
+	}
+}
+
+// controlKeysLoop asks as many controlkeys the user requires, until Done or Cancel is selected
+func controlKeysLoop(controlKeysPrompt string) ([]string, bool, error) {
+	const (
+		addCtrlKey = "Add control key"
+		doneMsg    = "Done"
+		cancelMsg  = "Cancel"
+	)
+
+	var controlKeys []string
+
+	for {
+		listDecision, err := prompts.CaptureList(
+			controlKeysPrompt, []string{addCtrlKey, doneMsg, cancelMsg},
```

Comment:
maybe remove the `done` option if the controlKeys list has length 0

Reply:
New approach is to keep asking for keys